server/container_create.go

```diff
@@ -384,6 +384,42 @@ func ensureSaneLogPath(logPath string) error {
 	return nil
 }
 
+// addSecretsBindMounts mounts user defined secrets to the container
+func addSecretsBindMounts(mountLabel, ctrRunDir, secretsFilePath string, specgen generate.Generator) error {
+	containerMounts := specgen.Spec().Mounts
+	// override the mount file path for testing purposes only
+	if secretsFilePath != "" {
+		testMounts, err := secretMounts(mountLabel, secretsFilePath, ctrRunDir, containerMounts)
```

Comment:
Why not pass the specgen into secretsMounts and have it handle all of the specgen.AddBindMount code rather then doing it three times.

Reply:
Won't it end up being the same thing? secretsMounts would have to loop through all three paths and bind the mounts.

Reply:
Sure but you wouldn't have cut and paste the same code in three places.

Reply:
agreed, I made a change. @rhatdan PTAL